builder/builder-source-archive.c

```diff
@@ -409,6 +410,22 @@ unzip (GFile   *dir,
   return res;
 }
 
+static gboolean
+unrpm (GFile   *dir,
+       const char *rpm_path,
+       GError **error)
+{
+  gboolean res;
+  const gchar *argv[] = { "sh", "-c", NULL, NULL };
+  char *unrpm_cmdline = g_strdup_printf("rpm2cpio %s | cpio -i -d", rpm_path);
+
+  argv[2] = unrpm_cmdline;
+  res = flatpak_spawnv (dir, NULL, error, argv);
+  g_free(unrpm_cmdline);
+
+  return res;
+}
+
```

Comment:
This looks easy enough. There's a few things we probably want to add though. I think we should check if the rpm content goes into the right prefix, to prevent people from mistakenly trying to bundle /usr content with the app. Do we need to check the architecture too ?